src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -195,7 +239,29 @@ public static KNNQueryBuilder fromXContent(XContentParser parser) throws IOExcep
             }
         }
 
-        KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(fieldName, ObjectsToFloats(vector), k, filter);
+        if (k < 0) {
```

Comment:
Can this section be refactored to reduce redundant code? I see similar checks happening above in ctor.